webapp/components/user_settings/email_notification_setting.jsx

```diff
@@ -37,9 +37,16 @@ export default class EmailNotificationSetting extends React.Component {
                 emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_FIFTEEN_MINUTES)
             };
         } else {
-            // otherwise, the default interval is immediately
+            /**
+             * otherwise, the default interval is immediately
+             *
+             * `emailInterval` in the backend is irrespective of whether email Bacthing is enabled or not.
+             * Hence this check is kept in the component to choose the default interval as `immediately` when
+             * `EnableEmailBatching` is false.
+             */
+
             this.state = {
-                emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_IMMEDIATE)
+                emailInterval: Preferences.INTERVAL_IMMEDIATE
```

Comment:
Agreed with your point. It will make the code more readable. I have made the changes.